ui/src/components/header/HeaderMenu.js

```diff
@@ -87,68 +92,103 @@ const HeaderMenuUserDiv = styled.div`
     margin-left: 15px;
 `;
 
-const HeaderMenu = (props) => {
-    let icon = 'notification';
-    const router = useRouter();
-    useEffect(() => {
-        props.getUserPendingMembers();
-    });
-    if (props.pending) {
-        if (Object.keys(props.pending).length !== 0) {
-            icon = 'notification-solid';
-        }
+const ReviewNotificationSpan = styled.span`
+    position: absolute;
+    top: 10px;
+    right: 90px;
+    width: 15px;
+    height: 15px;
+    background-color: red;
+    border-radius: 50px;
+`;
+
+class HeaderMenu extends React.Component {
+    constructor(props) {
+        super(props);
+        this.api = API();
+        this.state = {
+            roleGroupReviewFeatureFlag: false,
+        };
     }
-    let menuItems =
-        props.headerDetails &&
-        props.headerDetails.headerLinks &&
-        props.headerDetails.headerLinks.map((headerLink, idx) => {
-            return (
-                <MenuItemDiv key={idx}>
-                    <StyledAnchorDiv
-                        data-testid='menu'
-                        onClick={() =>
-                            window.open(headerLink.url, headerLink.target)
-                        }
-                    >
-                        {headerLink.title}
-                    </StyledAnchorDiv>
-                </MenuItemDiv>
-            );
+
+    onClickFunction(route) {
+        this.props.router.push(route, route);
+    }
+
+    componentDidMount() {
+        this.props.getUserPendingMembers();
+        this.props.getReviewRoles();
+        this.props.getReviewGroups();
+        this.api.getPageFeatureFlag('roleGroupReview').then((data) => {
+            this.setState({
+                roleGroupReviewFeatureFlag: data['roleGroupReviewFeatureFlag'],
```

Comment:
should we add this roleGroupReviewFeatureFlag to default-config.js

Reply:
done